### DISCUSSION TITLE
Include parse-url support for rabbitmq (amqp://) scheme

### DISCUSSION THREAD

**Discussion by itstimotei** [celery/celery#7698] [Q&A]
**Problem**: Singleton base are not supported if rabbitmq broker are used to assimilate the tasks.
**Error**: `Redis URL must specify one of the following schemes (redis://, rediss://, unix://)`

**Please include rabbitmq (amqp://) parse_url function from connection.py file -> line - approx 1194**

**If there is another way to use singleton with the rabbitmq broker, I would be very grateful to know. Thanks!**

recap: the problem is that when I use rabbitmq as a broker, I can't use Singleton property inside of decorator as a base, because parse_url method doesn't support that rabbitmq system

![Image 4](https://user-images.githubusercontent.com/86377196/183244246-2d5e7088-84be-4a45-a469-f852dca02069.png)

[a comment by an automated account is not shown]

**Comment by auvipy** [comment]
can you share a script which do not work as expected, so you asked for the change?

**Comment by itstimotei** [comment]
@auvipy Yes, here is the code:

**env.config file:**
```
CELERY_BROKER_HOST=redis://127.0.0.1:6379
```
 
**tasks.py file:**
```py
@app.task(base=Singleton, name="create_or_update_user")
def create_or_update(user_id: str):
    data = async_to_sync(update_or_create_user)(user_id=user_id)

    return data
```

**create_user.py file:**
```py
data = create_or_update.apply_async(
       args=[user_id],
       countdown=CELERY_TASKS_COUNTDOWN, 
)
```

**error**: `Redis URL must specify one of the following schemes (redis://, rediss://, unix://)`

**the problem is that when I use rabbitmq as a broker, I can't use Singleton property inside of decorator as a base, because parse_url method doesn't support the rabbitmq system**

**Comment by clokep** [comment]
@timotei-za I couldn't find the code you pasted an image of in either the celery or kombu repos. Could you provide a GitHub link? (Celery doesn't provide a Singleton class to my knowledge, are you using the celery-singleton project?)

**Comment by itstimotei** [comment]
@clokep yes, I'm using  celery-singleton library.

**Comment by itstimotei** [comment]
Is there any way to use Singleton as a task base if I use rabbitmq system as a broker?

**Comment by auvipy** [comment]
this issue shouldn't have report here at all, you should report here https://github.com/steinitzu/celery-singleton . thanks for wasting our time.
